### ISSUE TITLE
Upgrade Graal support to RC17+

### ISSUE THREAD

**Issue by ilopmar** [micronaut-projects/micronaut-core#1604]
Once this commit was introduced in Graal master branch https://github.com/oracle/graal/commit/b20dead4f4f0527dae948767992c82d4f43837f6, Graal support is broken in Micronaut because they introduced a few breaking changes. The workaround provided is adding:
```
--initialize-at-build-time=ch.qos.logback,com.fasterxml.jackson,fresh.graal,io.micronaut,io.netty,io.reactivex,org.reactivestreams,org.slf4j,org.yaml.snakeyaml,javax
```
To `native-image.properties`.

That makes some applications work again but:
- Almost all of our test applications work again: https://gitlab.com/micronaut-projects/micronaut-graal-tests/pipelines/59096627.
- I've opened an issue for RabbitMQ support: https://github.com/oracle/graal/issues/1211
- Management & Metrics: https://github.com/oracle/graal/issues/1212
- Security with JWT: https://github.com/oracle/graal/issues/1215
- In any case there are deprecation warnings that we need to address, but those are pending on another issue on their side regarding to lambdas.

I'll keep this issue open until everything is fixed and we can upgrade Micronaut to support Graal RC17+.

**Comment by ilopmar** [comment]
Added a new issue regarding the removal of the deprecation warnings: https://github.com/oracle/graal/issues/1217

**Comment by ilopmar** [comment]
Management & metrics application fixed: https://github.com/micronaut-graal-tests/micronaut-management-graal/commit/af3856bc61e62831119331da76fe7e3638fa9254

**Comment by ilopmar** [comment]
Issue with the lambdas: https://github.com/oracle/graal/issues/1218

**Comment by ilopmar** [comment]
All the issues regarding upgrading the tests applications are fixed and everything is back to green: https://gitlab.com/micronaut-projects/micronaut-graal-tests/pipelines/59403448

Keep in mind that this is still using the old (and now deprecated flags)
